lib/repeatable/expression/base.rb

```diff
@@ -24,6 +24,29 @@ def to_h
         )
       end
 
+      def union(other)
+        if other.is_a?(Union)
+          other.union(self)
+        else
+          Union.new(self, other)
+        end
+      end
+      alias + union
+
+      def intersection(other)
+        if other.is_a?(Intersection)
+          other.intersection(self)
+        else
+          Intersection.new(self, other)
+        end
+      end
+      alias | intersection
```

Comment:
I love the addition of these aliases!  What do you think about staying consistent with Ruby's [`Set` method names](http://ruby-doc.org/stdlib-2.3.1/libdoc/set/rdoc/Set.html#method-i-intersection) and using `&` for `intersection`?